indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentListUsedAction.java

```diff
@@ -34,16 +37,29 @@
   private final String dataSource;
 
   @JsonIgnore
-  private final Interval interval;
+  private final List<Interval> intervals;
 
   @JsonCreator
   public SegmentListUsedAction(
       @JsonProperty("dataSource") String dataSource,
-      @JsonProperty("interval") Interval interval
+      @JsonProperty("interval") Interval interval,
+      @JsonProperty("intervals") List<Interval> intervals
   )
   {
     this.dataSource = dataSource;
-    this.interval = interval;
+
+    Preconditions.checkArgument(
+        interval != null && intervals != null && !intervals.isEmpty(),
+        "pls specify intervals only"
+    );
+
+    List<Interval> theIntervals = null;
+    if (interval != null) {
+      theIntervals = ImmutableList.of(interval);
+    } else if (intervals != null && intervals.size() > 0) {
+      theIntervals = JodaUtils.condenseIntervals(intervals);
+    }
```

Comment:
This should probably prefer the new form… or just throw an exception if they are both provided. Nobody should be providing both.